### PR TITLE
Postgres support, JWT support and Some login providers

### PR DESCRIPTION
**Pull request by yuval-siev**

### CODE REVIEW

File: ui/src/app/page.tsx [hunk 1]
```diff
@@ -71,7 +71,7 @@ async function CrudShowcase() {
   return (
     <div className="w-full max-w-xs">
       {latestPost ? (
-        <p className="truncate">Your most recent post: {latestPost.name}</p>
+        <p className="truncate">Your most recent post: {"aaaa"}</p>
```
Comment (drorIvry) [inline ui/src/app/page.tsx:74]: Intentional?

File: ui/src/server/auth.ts [hunk 2]
```diff
@@ -38,16 +40,27 @@ declare module "next-auth" {
  */
 export const authOptions: NextAuthOptions = {
   callbacks: {
-    session: ({ session, user }) => ({
+    session: ({ session, token }) => ({
```
Comment (drorIvry) [inline ui/src/server/auth.ts:43]: I think we also need the iser obj in addition

File: ui/src/server/auth.ts [hunk 3]
```diff
@@ -38,16 +40,27 @@ declare module "next-auth" {
...
+  },
+  adapter: sqlDrizzleAdapter(db as unknown as NeonDatabase) as Adapter,
   providers: [
+    DiscordProvider({
+      clientId: process.env.DISCORD_CLIENT_ID ?? "",
+      clientSecret: process.env.DISCORD_CLIENT_SECRET ?? "",
+    }),
+    GitHubProvider({
+      clientId: process.env.GITHUB_ID ?? "",
+      clientSecret: process.env.GITHUB_SECRET ?? "",
```
Comment (drorIvry) [inline ui/src/server/auth.ts:62]: Env vars in t3 are done cia t3 env, checkout the env.mjs file

File: ui/src/server/db/schema.ts [hunk 4]
```diff
@@ -17,32 +17,31 @@ import { type AdapterAccount } from "next-auth/adapters";
...
  */
-export const mysqlTable = mysqlTableCreator((name) => `rego-ui_${name}`);
+export const pgTable = pgTableCreator((name) => `rego-ui_${name}`);
 
-export const posts = mysqlTable(
+export const posts = pgTable(
   "post",
   {
-    id: bigint("id", { mode: "number" }).primaryKey().autoincrement(),
+    id: bigint("id", { mode: "number" }).primaryKey(),
```
Comment (drorIvry) [inline ui/src/server/db/schema.ts:25]: I think you can define a CUID auto generated value

**Review by drorIvry — APPROVED** [review]

File: ui/src/app/page.tsx (hunk 1 above)
Comment (yuval-siev) [reply ui/src/app/page.tsx:74]: You wrote that when changing the db adapter :)

File: ui/src/server/db/schema.ts (hunk 4 above)
Comment (yuval-siev) [reply ui/src/server/db/schema.ts:25]: it's for the default post table.
it's a throwaway code anyway, I wouldn't put much into it

File: ui/src/server/auth.ts (hunk 3 above)
Comment (yuval-siev) [reply ui/src/server/auth.ts:62]: Added there as well

File: ui/src/server/auth.ts (hunk 2 above)
Comment (yuval-siev) [reply ui/src/server/auth.ts:43]: Can you elaborate?
I followed the code here:
https://create.t3.gg/en/usage/next-auth#usage-with-nextjs-middleware

Comment (drorIvry) [reply ui/src/server/auth.ts:43]: You don't need to remove the user obj
Just do this:

```javascript
callbacks: {
     session: ({ session, user }) => ({
     session: ({ session, token }) => ({
        ...session,
        user: {
          ...session.user,
         id: user.id,
         id: token.sub,
        },
      }),
```

Comment (drorIvry) [reply ui/src/server/auth.ts:43]: Hmm on that case ia looks great

File: ui/src/server/db/auth-adapter.ts [hunk 5]
```diff
@@ -0,0 +1,228 @@
+import type {
+  AdapterAccount,
```
Comment (drorIvry) [inline ui/src/server/db/auth-adapter.ts:2]: Didn't we already have that?

**Review by drorIvry — APPROVED** [review]
Awesome

**Pull request merged**